src/chart/options/options.cpp

```diff
@@ -233,10 +233,13 @@ uint64_t Options::generateMarkerInfoId()
 
 void Options::setAutoParameters()
 {
-	if (legend.get().isAuto()) {
-		LegendType tmp = legend.get();
-		tmp.setAuto(getAutoLegend());
-		legend = tmp;
+	if (auto [leg, w] = legend.get(0); leg.isAuto()) {
```

Comment:
I think we shouldn't have interpolated parameters here, so it would be safe to just use get(), and not checking the weight.